src/app/base/side-panel-context-new.tsx

```diff
@@ -0,0 +1,99 @@
+import type { ComponentType, PropsWithChildren, ReactElement } from "react";
+import { useContext, createContext, useState } from "react";
+
+// This new side panel context is meant to replace the old one over time. Since migrating all
+// panels will take some time, these two contexts will live together while the migrations are
+// being carried out.
+// TODO: Once the migrations are complete, remove the old side panel context.
+
+type SidePanelSize = "large" | "narrow" | "regular" | "wide";
+
+interface SidePanelState<TProps = Record<string, unknown>> {
+  isOpen: boolean;
+  title: string;
+  size: SidePanelSize;
+  component: ComponentType<TProps> | null;
+  props: TProps;
+}
+
+interface SidePanelActions {
+  open: <TProps extends Record<string, unknown> = Record<string, unknown>>(
+    component: ComponentType<TProps>,
+    title: string,
+    props?: TProps,
+    size?: SidePanelSize
+  ) => void;
+  close: () => void;
+  setSize: (size: SidePanelSize) => void;
```

Comment:
Just my opinion, but I think more descriptive names for returned variables would be nice, e.g. `openSidePanel`, `closeSidePanel`, `setSidePanelSize`. I know it's longer to type but I think it makes downstream code a bit more readable, especially in larger files when it's not obvious that these function calls are referring to the side panel